src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1333,36 +1336,38 @@ internal static bool IsCustomTaskType(this NamedTypeSymbol type, out NamedTypeSy
         /// </summary>
         internal static TypeSymbol NormalizeTaskTypes(this TypeSymbol type, CSharpCompilation compilation)
         {
-            NormalizeTaskTypesCore(compilation, ref type);
+            NormalizeTaskTypesInType(compilation, ref type);
             return type;
         }
 
         /// <summary>
         /// Replace Task-like types with Task types. Returns true if there were changes.
         /// </summary>
-        private static bool NormalizeTaskTypesCore(CSharpCompilation compilation, ref TypeSymbol type)
+        private static bool NormalizeTaskTypesInType(CSharpCompilation compilation, ref TypeSymbol type)
         {
             switch (type.Kind)
```

Comment:
Error types can have task-like types as type arguments and pointer types can refer to them, at least in error scenarios. Are you saying it is impossible to run into such types in this function? Then I would suggest adding cases for them with throw.   

---
In reply to: [74289912](https://github.com/dotnet/roslyn/pull/12671#discussion_r74289912) [](ancestors = 74289912,74289506,74288645,74285338)